docs-2.0/3.ngql-guide/7.general-query-statements/2.match.md

```diff
@@ -12,173 +14,630 @@ The syntax of `MATCH` is relatively more flexible compared with that of other qu
++---------------------------------------------------------------------------+
+Got 3 rows (time spent 4264/4976 us)
+```
+
+### Match multiple edges
+
+You can expand a pattern to match multiple edges in a path.
 
-nebula> MATCH (v:player {name: "Tim"}) RETURN v.age AS Age;
+```nGQL
+nebula> MATCH (v:player{name:"Tim Duncan"})-[]->(v2)<-[e:serve]-(v3) RETURN v2, v3;
++------------------------------------+-----------------------------------------------------------+
+| v2                                 | v3                                                        |
++------------------------------------+-----------------------------------------------------------+
+| ("player204" :team{name: "Spurs"}) | ("player101" :player{name: "Tony Parker", age: 36})       |
++------------------------------------+-----------------------------------------------------------+
+| ("player204" :team{name: "Spurs"}) | ("player102" :player{name: "LaMarcus Aldridge", age: 33}) |
++------------------------------------+-----------------------------------------------------------+
+| ("player204" :team{name: "Spurs"}) | ("player103" :player{age: 32, name: "Rudy Gay"})          |
++------------------------------------+-----------------------------------------------------------+
+...
+```
+
+### Match fixed-length paths
+
+To match a fixed-length path, use the `:<edge_type>*<hop>` pattern.
+
+```nGQL
+nebula> MATCH p=(v:player{name:"Tim Duncan"})-[e:like*2]->(v2) RETURN DISTINCT v2 AS Friends;
++-----------------------------------------------------------+
+| Friends                                                   |
++-----------------------------------------------------------+
+| ("player100" :player{name: "Tim Duncan", age: 42})        |
++-----------------------------------------------------------+
+| ("player102" :player{name: "LaMarcus Aldridge", age: 33}) |
++-----------------------------------------------------------+
+| ("player125" :player{name: "Manu Ginobili", age: 41})     |
++-----------------------------------------------------------+
+Got 3 rows (time spent 4863/5591 us)
+```
+
+### Match variable-length paths
+
+You can use the `:<edge_type>*[minHop]..<maxHop>` pattern to match variable-length paths.
+
+|Parameter|Description|
+|-|-|
+|`minHop`|Optional. Represents the minimum length of the path. The default value is 1.|
+|`maxHop`|Required. Represents the maximum length of the path.|
+
+> **OpenCypher compatibility**
+>
+> * In nGQL, `maxHop` is required. And `..` cannot be omitted.
+> * In openCypher, `maxHop` is optional and default to infinity. When no bounds are given `..` can be omitted.
+
+```nGQL
+nebula> MATCH p=(v:player{name:"Tim Duncan"})-[e:like*1..3]->(v2) \
+        RETURN v2 AS Friends;
++-----------------------------------------------------------+
+| Friends                                                   |
++-----------------------------------------------------------+
+| ("player101" :player{age: 36, name: "Tony Parker"})       |
```

Comment:
the properties are in a different order. it is right, right?

Reply:
Yes, properties are in a list, unordered.